lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4818,7 +4818,8 @@ namespace Js
 #endif
         }
 
-        if (result == FALSE && (propertyOperationFlags & PropertyOperation_StrictMode))
+        if (result == FALSE && 
+            (propertyOperationFlags & (PropertyOperation_StrictMode | PropertyOperation_ThrowOnDeleteIfNotConfig)))
```

Comment:
This looks reasonable to me, but it does make me wonder whether there's some other scenario somewhere that might directly call into `someArray->DeleteItem` and expect that it throws when appropriate; a quick look around showed code like this: https://github.com/Microsoft/ChakraCore/blob/master/lib/Runtime/Library/JavascriptArray.cpp#L5649 which I think converts the `false` return into throwing explicitly, but seems to expect the typed array will do something with the flag it is given rather than completely ignoring it. Do you know if there are any other potential scenarios?

Reply:
+1

Not sure this is the right place for this check. When `result == FALSE`, it might be for another reason other than not `configurable`.

Reply:
`result` looks like it always is returned by `JavascriptOperators::DeleteProperty`. What other reason could it return false for?

Reply:
It looks to me like result is only false in the non-configurable case, so this should work.

Like Jimmy mentioned, various other methods invite misuse because they accept a PropertyOperationFlags parameter but don't pay attention to PropertyOperation_ThrowOnDeleteIfNotConfig. It seems somewhat inconsistent; some methods like PathTypeHandlerBase::DeleteProperty do respect that flag, but others don't. I'm not sure if it's worth trying to track them all down at this point.

---
In reply to: [173333391](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173333391) [](ancestors = 173333391)

Reply:
After all, it seems that it's better to have this handling in TypedArrayBase::DeleteItem as we discussed before :)

---
In reply to: [173024703](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173024703) [](ancestors = 173024703)